src/main/java/com/exasol/adapter/dynamodb/documentpath/DocumentPathExpression.java

```diff
@@ -0,0 +1,96 @@
+package com.exasol.adapter.dynamodb.documentpath;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Using this class a path through a document can be expressed.
+ */
+public class DocumentPathExpression {
+    private final List<PathSegment> path;
+
+    private DocumentPathExpression(final List<PathSegment> path) {
+        this.path = path;
+    }
+
+    List<PathSegment> getPath() {
+        return this.path;
+    }
+
+    /**
+     * Creates a sub path starting at startIndex until endIndex.
```

Comment:
```suggestion
     * Creates a subpath from startIndex (inclusive) til endIndex (exclusive).
```
Or both of them are included?